AmplifyTestCommon/Mocks/MockAPICategoryPlugin.swift

```diff
@@ -9,7 +9,7 @@ import Amplify
 import Combine
 import Foundation
 
-class MockAPICategoryPlugin: MessageReporter, APICategoryPlugin {
+class MockAPICategoryPlugin: MessageReporter, APICategoryPlugin, APICategoryReachabilityBehavior {
```

Comment:
we're added conformance to the protocol but no new methods implemented in the mock, so casting a mock API to `APICategoryReachabilityBehavior` works, but what happens when it tries to use methods from there? 

does `networkReachabilityPublisher = try reachability.reachabilityPublisher()` fall into the catch block? Is there a mock publisher we can return?

Reply:
I found out there is one implemented! (https://github.com/aws-amplify/amplify-ios/blob/1d96b55691de534324f8f1256bc6517cad074982/AmplifyTestCommon/Mocks/MockAPICategoryPlugin.swift#L111)

Reply:
can we test returning a mocked response? like https://developer.apple.com/documentation/combine/just
```
return Just(ReachabilityUpdate.active)
```